File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-0.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        def rec(i,j):
            for k in range(len(matrix[0])):
                if matrix[i][k] != 0:
                    matrix[i][k] = None
            for k in range(len(matrix)):
                if matrix[k][j] != 0:
                    matrix[k][j] = None
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == 0:
                    rec(i,j)
        
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == None:
                    matrix[i][j] = 0

File: Data Structures & Algorithms/search-2d-matrix/submission-0.js
class Solution {
    /**
     * @param {number[][]} matrix
     * @param {number} target
     * @return {boolean}
     */
    searchMatrix(matrix, target) {
       let  m = matrix.length;
       let  n = matrix[0].length;
       for(let c of matrix){
         if(target >= c[0] || target <= c[n-1]){
            for(let n of c){
                if(target == n )return true
            }
         }
       }
       return false
    }
}
